test/mapmatch.cc

```diff
@@ -108,6 +111,40 @@ const auto conf = json_to_pt(R"({
     }
   })");
 
+struct api_tester {
```

Comment:
to properly test this i rewrote the test. now we run first a multileg route, then a multileg map match. and we compare that we get the same number of routes/legs/elapsed time per leg. basically it says if you do the input right, the same request to routing and map matching should give the same answer.